students/ravi/Session03/list_lab.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+#Series 1
+
+"""Create a list that contains “Apples”, “Pears”, “Oranges” 
+and “Peaches”."""
+fruits = ['Apples','Pears','Oranges','Peaches']
+
+#Display the list
+print(fruits)
+
+#Ask the user for another fruit and add it to the end of the list.
+user_fruit = input("add a fruit of your choice > ")
+fruits.append(user_fruit)
+
+#Display the list
+print(fruits)
+
+"""Ask the user for a number and display the number 
+back to the user and the fruit corresponding to that number 
+(on a 1-is-first basis). """
+user_number = input("enter a number from 1 to 5 > ")
+print (user_number," ",fruits[int(user_number)-1])
+
+"""Add another fruit to the beginning of the list using “+” 
+and display the list."""
+fruits = ["Mangoes"]+fruits
+print(fruits)
+
+"""Add another fruit to the beginning of the list using 
+insert() and display the list."""
+fruits.insert(0,'Grapes')
+print(fruits)
+
+#Display all the fruits that begin with “P”, using a for loop.
+for i in range(len(fruits)):
+    chosen_fruit=fruits[i]
+    if chosen_fruit[0]=="P":
+        print(chosen_fruit)
+
+#Series 2
+
+#Display the list.
+print(fruits)
+
+#Remove the last fruit from the list.
+fruits=fruits[:-1]
+
+#Display the list.
+print(fruits)
+
+#Ask the user for a fruit to delete, find it and delete it.
+user_removed_fruit = input("enter a fruit to remove > ")
+fruits_to_remove=[]
+for i in range(len(fruits)):
+    if fruits[i]==user_removed_fruit:
+        fruits_to_remove.append(fruits[i])
+new_list=[x for x in fruits if x not in fruits_to_remove]
+print(new_list)
+
+'''(Bonus: Multiply the list times two. Keep asking 
+until a match is found. Once found, delete all occurrences.)'''
+
+new_list=new_list*2
+fruits_to_remove=[]
+while fruits_to_remove==[]:
+    user_removed_fruit = input("enter a fruit to remove > ")
+    for i in range(len(new_list)):
+        if new_list[i]==user_removed_fruit:
+            fruits_to_remove.append(new_list[i])
+newer_list=[x for x in new_list if x not in fruits_to_remove]
+print(newer_list)
```

Comment:
Nice comprehension. Note that this can be done in one loop:
```
user_removed_fruit = input("enter a fruit to remove > ")
while user_removed_fruit in new_list:
    new_list.remove(user_removed_fruit)
```
alternatively, if you want a new list, you can do this (which only loops through once):
```
new_list = []
for fruit in d_fruits * 2:
    if fruit != user_removed_fruit:
        new_list.append(fruit)
```